boards: xtensa: Introduce esp32s3_luatos_core board

feilongfl opened the pull request:
The ESP32S3-CORE development board is a compact board designed based on Espressif ESP32-S3. The board comes equipped with a 2.4GHz antenna and supports both Wi-Fi and Bluetooth functionalities. For more information, check [ESP32S3-CORE](https://wiki.luatos.com/chips/esp32s3/board.html) (chinese)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10269596/9723114d-f185-4377-987e-513ebb120f59)

str4t0m reviewed (commented):
"Core board SoC-xy" seems to be a very common naming scheme for very elementary boards and searching online shops briefly I found multiple different esp32s3 boards called core board. So having such an unspecific name might lead to confusion in the future.

What about adding a prefix to the board name to identify the manufacturer? I not am sure about the manufacturer but something like "airm2m", "luat", or luatos.

sylvioalves commented:
> 

I have the same feeling. Perhaps `esp32s3_luatos_core` or `luatos_esp32s3_core` would be interesting.

feilongfl commented:
> > 
> 
> I have the same feeling. Perhaps `esp32s3_luatos_core` or `luatos_esp32s3_core` would be interesting.

I prefer `esp32s3_luatos_core`, I changed the board name to this.

kartben commented:
> > > 
> > 
> > I have the same feeling. Perhaps `esp32s3_luatos_core` or `luatos_esp32s3_core` would be interesting.
> 
> I prefer `esp32s3_luatos_core`, I changed the board name to this.
> 

This will need to be updated in the commit message too, please.

feilongfl commented:
pipeline failed with command: `west build -p -b esp32s3_luatos_core -T tests/kernel/fpu_sharing/generic/kernel.fpu_sharing.generic.xtensa`

cause by #61455 
seems like we need implement `arch_float_disable` and `arch_float_enable` functions.

marekmatej commented on boards/xtensa/esp32s3_luatos_core/doc/index.rst:
```diff
@@ -0,0 +1,284 @@
+Overview
+********
+
+The ESP32S3-LUATOS-CORE development board is a compact board designed based on Espressif ESP32-S3.
```
Please fix or rephrase - 'designed' OR 'based'

marekmatej commented on boards/xtensa/esp32s3_luatos_core/esp32s3_luatos_core.dts:
```diff
@@ -0,0 +1,183 @@
+};
+
+&usb_serial {
+	status = "okay";
```
Is it necessary to enable usb_serial by default? There is CH343 driver on-board.

feilongfl replied:
Reference: https://wiki.luatos.com/chips/esp32s3/hardware.html

There is a switch (S2) on the development board to control the USB switch (S1),  
and the control PC is directly connected to ESP32 or connected through ch343.  

I don't understand the different of CH343 on the programming and debugging functions,   
but I think enable usb_serial can make it work in either mode.

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10269596/cbdff405-6f69-45eb-8e6b-65957977285d)

feilongfl replied:
I realized I just misread that this is a usb serial port and not a debug port.

There is a switch (S2) on the development board to control the USB switch (S1),  
and the control PC is directly connected to ESP32 or connected through ch343.  

![image](https://github.com/zephyrproject-rtos/zephyr/assets/10269596/cbdff405-6f69-45eb-8e6b-65957977285d)

So when the usb switch enables CH343 you are right, we don't need to use usb_serial.
But when the switch directly disables CH343, usb_serial needs to be turned on.

feilongfl replied:
Perhaps it would be better to provide two dts files corresponding to whether CH343 is enabled or not?

``` kconfig
# Kconfig.board

# ESP32S3 Core board configuration
config BOARD_ESP32S3_LUATOS_CORE_NOCH343
	bool "ESP32S3 LuatOS Core Board with ch343 disabled"
	depends on SOC_SERIES_ESP32S3

config BOARD_ESP32S3_LUATOS_CORE_CH343
	bool "ESP32S3 LuatOS Core Board with ch343 enabled"
	depends on SOC_SERIES_ESP32S3

```

marekmatej replied:
I commented on this because, on (both) of my `esp32s3_luatos` boards, the `S2` switch default position is `CH343`. So I assume if CDC is needed, it would be enabled at the application overlay.

feilongfl committed:
boards: xtensa: Introduce esp32s3_luatos_core board

ESP32S3 LuatOS core is a dev board with esp32-s3.
It has similar functions to esp32s3 devkitm, but smaller in size.

Signed-off-by: YuLong Yao <feilongphone@gmail.com>

marekmatej reviewed (approved):
Thank you, nice job!
